drivers/wifi/simplelink_support.c

```diff
@@ -0,0 +1,616 @@
+
+	/* Disable ipv6 */
+	if_bitmap = !(SL_NETCFG_IF_IPV6_STA_LOCAL |
+		      SL_NETCFG_IF_IPV6_STA_GLOBAL);
+	retval = sl_NetCfgSet(SL_NETCFG_IF, SL_NETCFG_IF_STATE,
+			      sizeof(if_bitmap),
+			      (const unsigned char *)&if_bitmap);
+	ASSERT_ON_ERROR(retval, NETAPP_ERROR);
+
+	/* Configure scan parameters to default */
+	scan_default.ChannelsMask = CHANNEL_MASK_ALL;
+	scan_default.RssiThershold = RSSI_TH_MAX;
+
+	retval = sl_WlanSet(SL_WLAN_CFG_GENERAL_PARAM_ID,
+			    SL_WLAN_GENERAL_PARAM_OPT_SCAN_PARAMS,
+			    sizeof(scan_default), (u8_t *)&scan_default);
+	ASSERT_ON_ERROR(retval, WLAN_ERROR);
+
+	/* Disable scans: In other words, use "one-shot" scanning */
+	config_opt = SL_WLAN_SCAN_POLICY(0, 0);
+	retval = sl_WlanPolicySet(SL_WLAN_POLICY_SCAN, config_opt, NULL, 0);
+	ASSERT_ON_ERROR(retval, WLAN_ERROR);
+
+	/* Set TX power lvl to max */
+	power = 0;
+	retval = sl_WlanSet(SL_WLAN_CFG_GENERAL_PARAM_ID,
+			    SL_WLAN_GENERAL_PARAM_OPT_STA_TX_POWER, 1,
+			    (u8_t *)&power);
+	ASSERT_ON_ERROR(retval, WLAN_ERROR);
+
+	/* Set NWP Power policy to 'normal' */
+	retval = sl_WlanPolicySet(SL_WLAN_POLICY_PM, SL_WLAN_NORMAL_POLICY,
+				  NULL, 0);
+	ASSERT_ON_ERROR(retval, WLAN_ERROR);
+
+	/* Unregister mDNS services */
+	retval = sl_NetAppMDNSUnRegisterService(0, 0, 0);
+	ASSERT_ON_ERROR(retval, NETAPP_ERROR);
+
+	/* Remove all 64 RX filters (8*8) */
+	memset(rx_filterid_mask.FilterBitmap, 0xFF, 8);
+
+	retval = sl_WlanSet(SL_WLAN_RX_FILTERS_ID, SL_WLAN_RX_FILTER_REMOVE,
+			    sizeof(SlWlanRxFilterOperationCommandBuff_t),
+			    (u8_t *)&rx_filterid_mask);
+	ASSERT_ON_ERROR(retval, WLAN_ERROR);
+
+	/* Set NWP role as STA */
+	retval = sl_WlanSetMode(ROLE_STA);
+	ASSERT_ON_ERROR(retval, WLAN_ERROR);
+
+	/* For changes to take affect, we restart the NWP */
+	retval = sl_Stop(0xFF);
+	ASSERT_ON_ERROR(retval, DEVICE_ERROR);
+
+	mode = sl_Start(0, 0, 0);
+	ASSERT_ON_ERROR(mode, DEVICE_ERROR);
+
+	if (mode != ROLE_STA) {
+		SYS_LOG_ERR("Failed to configure device to it's default state");
+		retval = -1;
+	} else {
+		nwp.role = ROLE_STA;
+		SET_STATUS_BIT(nwp.status, STATUS_BIT_NWP_INIT);
+		retval = 0;
+	}
+
+	return retval;
+}
+
+/**
+ *  @brief SimpleLinkWlanEventHandler
+ *
+ *  This handler gets called whenever a WLAN event is reported
+ *  by the host driver / NWP.
+ *
+ *  @note See the CC3120/CC3220 NWP programmer's guide (SWRU455)
+ *	  sections 4.3.4, 4.4.5 and 4.5.5.
+ */
+void SimpleLinkWlanEventHandler(SlWlanEvent_t *wlan_event)
+{
+	SlWlanEventDisconnect_t	 *event_data = NULL;
+
+	if (!wlan_event) {
+		return;
+	}
+
+	switch (wlan_event->Id) {
+	case SL_WLAN_EVENT_CONNECT:
+		SET_STATUS_BIT(nwp.status, STATUS_BIT_CONNECTION);
+
+		/* Store new connection SSID and BSSID: */
+		memcpy(sl_conn.ssid, wlan_event->Data.Connect.SsidName,
+		       wlan_event->Data.Connect.SsidLen);
+		memcpy(sl_conn.bssid, wlan_event->Data.Connect.Bssid,
+		       BSSID_LEN_MAX);
+
+		SYS_LOG_INF("\n[WLAN EVENT] STA Connected to the AP: %s, "
```

Comment:
No need to put \n to syslog prints, the output will look funny in practice if you have \n in the output.

Reply:
I generally agree with the principle of not putting newlines in log print out, especially if the underlying logger appends a '\n'.
But in this case, the SYS_LOG message prefix is so long  (eg: "dev/simplelink [INF] SimpleLinkWlanEventHandler: ") that the important information (the [WLAN EVENT] + SSID and IP addresses) end up wrapping around the console (typically set around 40 or so characters), so that it actually becomes very difficult to read.
Adding the '\n' at the start keeps the SSID names and IP addresses within the same line, and so much more readable.